Gems/Atom/Feature/Common/Code/Source/SkinnedMesh/SkinnedMeshInputBuffers.cpp

```diff
@@ -180,7 +182,8 @@ namespace AZ
         void SkinnedMeshInputLod::TrackStaticBufferViews(uint32_t meshIndex)
         {
             SkinnedSubMeshProperties& skinnedSubMesh = m_meshes[meshIndex];
-            const RPI::ModelLodAsset::Mesh& modelLodAssetMesh = m_modelLodAsset->GetMeshes()[meshIndex];
+            const auto modelLodAssetMeshes = m_modelLodAsset->GetMeshes();
+            const RPI::ModelLodAsset::Mesh& modelLodAssetMesh = modelLodAssetMeshes[meshIndex];
```

Comment:
The reason GCC sees this code as a potential dangling reference is because the [GetMeshes](https://github.com/o3de/o3de/blob/9730572ef3909d48e8c945fbdd29c0e6b7e5ed6f/Gems/Atom/RPI/Code/Include/Atom/RPI.Public/Model/ModelLod.h#L96) function returns an `AZStd::span<const Mesh>`.

Now the compiler doesn't know the semantics of the `AZStd::span` span type, so it can't know that invoking the [T& operator[]](https://github.com/o3de/o3de/blob/9730572ef3909d48e8c945fbdd29c0e6b7e5ed6f/Code/Framework/AzCore/AzCore/std/containers/span.h#L156-L157) function is returning a reference to a view-like type which is safe.

It only knows that the type being returned is a value type and then a function being invoked on it is then returning a reference to that value.